sys/include/fmt.h

```diff
@@ -76,6 +91,23 @@ size_t fmt_byte_hex(char *out, uint8_t byte);
  */
 size_t fmt_bytes_hex_reverse(char *out, const uint8_t *ptr, size_t n);
 
+/**
+ * @brief Converts a sequence of hex bytes to an array of bytes
+ *
+ * The sequence of hex characters must have an even length:
+ * 2 hex character => 1 byte. If the sequence of hex has an odd length, this
+ * function returns 0 and an empty @p out.
+ *
+ * The hex characters sequence must contain valid hexadecimal characters
+ * otherwise the result in @p out is undefined.
+ *
+ * @param[out] out  Pointer to converted bytes, or NULL
+ * @param[in]  hex  Pointer to input buffer
+ * @returns    strlen(hex) / 2 for a valid sequence of hex bytes
```

Comment:
Personal preference, but I would prefer another specification of what a "valid sequence" is here (e.g. "`strlen(hex) / 2 when length of @p hex was even`"

Reply:
ok, done